infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/events/BrokerService.java

```diff
@@ -98,18 +98,29 @@ private void handle(BrokerStatusChangedEvent event) {
       LOG.error("Broker event skipped due to illegal content: {}", event);
       return;
     }
-    eventService.publish(new BrokerEvent(event, parseTooling(encodedTooling)));
-  }
-
-  @Nullable
-  private List<ChePlugin> parseTooling(String toolingString) {
-    if (!isNullOrEmpty(toolingString)) {
+    BrokerEvent brokerEvent;
+    List<ChePlugin> tooling = null;
+    if (!isNullOrEmpty(encodedTooling)) {
       try {
-        return objectMapper.readValue(toolingString, new TypeReference<List<ChePlugin>>() {});
+        tooling = parseTooling(encodedTooling);
       } catch (IOException e) {
-        LOG.error("Parsing Che plugin broker event failed. Error: " + e.getMessage(), e);
+        if (isNullOrEmpty(event.getError())) {
+          event.withError(e.getMessage());
+        }
       }
     }
-    return null;
+    brokerEvent = new BrokerEvent(event, tooling);
+    eventService.publish(brokerEvent);
+  }
+
+  @Nullable
```

Comment:
It can not return null value anymore, right?

Reply:
changed